src/useNotification.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+import * as thisModule from './useNotification';
+
+export async function requestPermission(setPermission) {
+  const permissionFromUser = await Notification.requestPermission();
+
+  setPermission(permissionFromUser);
+}
+
+const useNotification = (title: string, options: NotificationOptions): (() => void) => {
+  const [permission, setPermission] = useState(
+    Notification && Notification.permission ? Notification.permission : 'denied'
+  );
+
+  useEffect(() => {
+    if (permission !== 'denied' && permission !== 'granted') {
+      thisModule.requestPermission(setPermission);
```

Comment:
Interesting, why do you use `thisModule.requestPermission` instead of just `requestPermission`?

Reply:
This allows me to mock and spy on `requestPermission` in the tests. If you know of another method to accomplish this I am very interested :)